src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      const isCalculatorMethod = Object.keys(this).some((method) => this[method] === callback);
+
+      if (isCalculatorMethod && typeof callback === 'function') {
+        callback.call(this, number);
+
+        return this;
+      }
+    },
```

Comment:
The `operate` method does not handle the case where the callback is not a function or is not a calculator method. It should provide feedback or throw an error to inform the user of incorrect usage.